src/sqlfluff/dialects/dialect_databricks.py

```diff
@@ -377,3 +378,58 @@ class AliasExpressionSegment(sparksql.AliasExpressionSegment):
             ),
         ),
     )
+
+
+class GeneratedColumnDefinitionSegment(sparksql.GeneratedColumnDefinitionSegment):
+    """A generated column definition, e.g. for CREATE TABLE or ALTER TABLE.
+
+    https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html
+    """
+
+    match_grammar: Matchable = Sequence(
+        Ref("SingleIdentifierGrammar"),  # Column name
+        Ref("DatatypeSegment"),  # Column type
+        Bracketed(Anything(), optional=True),  # For types like VARCHAR(100)
```

Comment:
```suggestion
        Bracketed(Anything(), optional=True),  # For types like DECIMAL(3, 2)
```
Databricks has no `VARCHAR` type.

Reply:
I believe DataBricks does let you define columns as varchar, but just converts it to a string under the hood.

Reply:
Changed it as suggested anyway, as I do agree that using VARCHAR in Databricks is quite unusual.